File: fragmentation-core-androidx/src/main/java/me/yokeyword/fragmentation/anim/DefaultVerticalAnimator.java
package me.yokeyword.fragmentation.anim;

import android.os.Parcel;
import android.os.Parcelable;

import me.yokeyword.fragmentation.R;

/**
 * Created by YoKeyword on 16/2/5.
 */
public class DefaultVerticalAnimator extends FragmentAnimator implements Parcelable {

    public DefaultVerticalAnimator() {
        mEnter = R.anim.v_fragment_enter;
        mExit = R.anim.v_fragment_exit;
        mPopEnter = R.anim.v_fragment_pop_enter;
        mPopExit = R.anim.v_fragment_pop_exit;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        super.writeToParcel(dest, flags);
    }

    private DefaultVerticalAnimator(Parcel in) {
        super(in);
    }

    public static final Creator<DefaultVerticalAnimator> CREATOR = new Creator<DefaultVerticalAnimator>() {
        @Override
        public DefaultVerticalAnimator createFromParcel(Parcel source) {
            return new DefaultVerticalAnimator(source);
        }

        @Override
        public DefaultVerticalAnimator[] newArray(int size) {
            return new DefaultVerticalAnimator[size];
        }
    };
}

File: fragmentation-core-androidx/src/main/java/me/yokeyword/fragmentation/anim/DefaultVerticalAnimator.kt
package me.yokeyword.fragmentation.anim

import android.os.Parcel
import android.os.Parcelable

import me.yokeyword.fragmentation.R

/**
 * Created by YoKeyword on 16/2/5.
 */
class DefaultVerticalAnimator : FragmentAnimator, Parcelable {

    constructor() {
        enterAnim = R.anim.v_fragment_enter
        exitAnim = R.anim.v_fragment_exit
        popEnterAnim = R.anim.v_fragment_pop_enter
        popExitAnim = R.anim.v_fragment_pop_exit
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        super.writeToParcel(dest, flags)
    }

    private constructor(`in`: Parcel) : super(`in`)

    companion object {

        @JvmField
        val CREATOR: Parcelable.Creator<DefaultVerticalAnimator> = object : Parcelable.Creator<DefaultVerticalAnimator> {
            override fun createFromParcel(source: Parcel): DefaultVerticalAnimator {
                return DefaultVerticalAnimator(source)
            }

            override fun newArray(size: Int): Array<DefaultVerticalAnimator?> {
                return arrayOfNulls(size)
            }
        }
    }
}
